core/spec/models/spree/store_credit_spec.rb

```diff
@@ -754,19 +754,34 @@
           it "saves the user's total store credit in the event" do
             expect(subject.store_credit_events.first.user_total_amount).to eq store_credit_amount
           end
+
+          it "saves the user's unused store credit in the event" do
+            expect(subject.store_credit_events.first.amount_remaining).to eq store_credit_amount
+          end
         end
 
         context "user has multiple store credits" do
           let(:store_credit_amount)            { 100.0 }
           let(:additional_store_credit_amount) { 200.0 }
-
           let(:user)                           { create(:user) }
-          let!(:store_credit)                  { create(:store_credit, user: user, amount: store_credit_amount) }
 
-          subject { create(:store_credit, user: user, amount: additional_store_credit_amount) }
+          let!(:store_credits) do
+            [
+              create(:store_credit, user: user, amount: store_credit_amount),
+              create(:store_credit, user: user, amount: additional_store_credit_amount)
+            ]
+          end
+
+          subject { store_credits.flat_map(&:store_credit_events) }
 
           it "saves the user's total store credit in the event" do
-            expect(subject.store_credit_events.first.user_total_amount).to eq (store_credit_amount + additional_store_credit_amount)
+            expect(subject.first.user_total_amount).to eq store_credit_amount
+            expect(subject.last.user_total_amount).to eq(store_credit_amount + additional_store_credit_amount)
+          end
+
+          it "saves the user's unused store credit in the event" do
+            expect(subject.first.amount_remaining).to eq store_credit_amount
+            expect(subject.last.amount_remaining).to eq additional_store_credit_amount
```

Comment:
@stewart you added assertions here and above that reference `subject.last` that don't appear in the still-open PR for Solidus: https://github.com/solidusio/solidus/pull/1512/files#diff-b3fa806cb8560b7b39d6791b65809f3a

Do you recall why? Should those assertions be added to the upstream PR? I'd like to see it merged so we can eventually remove this backport.